### DISCUSSION TITLE
Creating a Plugin - Not getting executed because not part of flavor?

### DISCUSSION THREAD

**Discussion by BrunoJuchli** [oxsecurity/megalinter#2135] [Q&A]
Hi there
I'm trying to create a plugin "for dotnet" to use the `dotnet jb` command (JetBrains ReSharper Command Line).
The plugin gets successfully loaded:

> [Plugins] Loaded plugin descriptor /megalinter-descriptors/dotnet-jb.megalinter-descriptor.yml from file://.github/mega-linter-plugin-dotnet-jb/dotnet-jb.megalinter-descriptor.yml
> [Plugins] Successful initialization of RESHARPER plugins

However I later get this error: 

> MegaLinter flavor [dotnet] does not contain linters RESHARPER.
As they are not available in this docker image, they will not be processed
To solve this problem, please either:
>- use default flavor oxsecurity/megalinter
>- add ignored linters in DISABLE or DISABLE_LINTERS variables in your .mega-linter.yml config file located in your root directory
>- ignore this message by setting config variable FLAVOR_SUGGESTIONS to false

I tried adding `FLAVOR_SUGGESTIONS: false` which didn't help. I guess it's only hiding the warning message, not actually changing the behavior?

What's necessary for mega-linter to pick up the linter in the plugin descriptor and considering it available? Or can plugins only be used with the default flavor?

the plugin descriptor file looks like this:

    descriptor_id: RESHARPER
    descriptor_label: "dotnet jetbrains resharper"
    descriptor_type: tooling_format
    descriptor_flavors:
      - dotnet
    file_extensions:
      - ".cs"
    linters:
      - linter_name: jb
        name: RESHARPER
        is_formatter: true
        linter_url: https://www-jetbrains-com.translate.goog/help/resharper/ReSharper_Command_Line_Tools.html
        cli_lint_mode: project
        cli_lint_extra_args:
          - "inspectcode"
          - "cleanupcode"
        cli_lint_fix_arg_name: "--megalinter-fix-flag" # Workaround for PythonBlackOrIsortLinter class behavior
        cli_lint_fix_remove_args:
          - "inspectcode"
        cli_lint_check_remove_args:
          - "cleanupcode"
        cli_lint_extra_args_after:
          - "src/smartMeCloud.sln"
          - "--profile:\"CI\""
        linter_version_cache: "000"
        examples:
          - "jb cleanupcode src/smartMeCloud.sln --profile:\"CI\"" # Fix
         install:
          dockerfile:
            - RUN /usr/share/dotnet/dotnet tool install -g JetBrains.ReSharper.GlobalTools --version 2022.3

**Comment by nvuillam** [comment]
Can you try to host the plugin online ? Maybe it would work better
There may also be an error because of the flavor... please can you try with full megalinter to see the difference ?

**Reply by BrunoJuchli** [reply]
@nvuillam 
Thanks for the tips. Switching to non-flavored mega-linter seems to help, since now I get another error:

> ...
> File "/usr/local/lib/python3.10/site-packages/megalinter-0.1-py3.10.egg/megalinter/Linter.py", line 846, in execute_lint_command
    process = subprocess.run(
> ...
> FileNotFoundError: [Errno 2] No such file or directory: 'jb'

Which I interpret as mega-linter is now trying to execute the plugin-linter.

Would be nice if the flavored mega-linter would support plugins, too, though.
